src/engine_redraw.c

```diff
@@ -150,7 +150,7 @@ static void draw_creature_view_icons(struct Thing* creatng)
         if (!creature_instance_is_available(creatng, cctrl->active_instance_id))
         {
             x = MyScreenWidth - (scale_value_by_horizontal_resolution(148) / 4);
-            draw_gui_panel_sprite_left(x, y, ps_units_per_px, instance_button_init[cctrl->active_instance_id].symbol_spridx);
+            draw_gui_panel_sprite_left(x, y, ps_units_per_px, instance_button_init[(short)cctrl->active_instance_id].symbol_spridx);
```

Comment:
`cctrl->active_instance_id % 48` probably better as `instance_button_init` has a length of 48. Casting to short still allows out of bounds.